overlord/configstate/configcore/ctrlaltdel.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/logger"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+	"github.com/snapcore/snapd/sysconfig"
+	"github.com/snapcore/snapd/systemd"
+)
+
+// Systemd unit name to control ctrl-alt-del behaviour
+const (
+	ctrlAltDelTarget = "ctrl-alt-del.target"
+)
+
+// Supported actions for core.system.ctrl-alt-del-action
+const (
+	ctrlAltDelReboot = "reboot"
+	ctrlAltDelNone   = "none"
+)
+
+func init() {
+	supportedConfigurations["core.system.ctrl-alt-del-action"] = true
+}
+
+type sysdCtrlAltDelLogger struct{}
+
+func (l *sysdCtrlAltDelLogger) Notify(status string) {
+	logger.Noticef("CtrlAltDel handler: %s", status)
+}
+
+// switchCtrlAltDelAction configures the systemd handling of the special
+// ctrl-alt-del keyboard sequence. This function supports configuring
+// systemd to trigger a reboot, or ignore the key sequence.
+func switchCtrlAltDelAction(action string, opts *fsOnlyContext) error {
+	validAction := map[string]bool{
+		ctrlAltDelReboot: true,
+		ctrlAltDelNone:   true,
+	}
+	if !validAction[action] {
+		return fmt.Errorf("invalid action %q supplied for system.ctrl-alt-del-action option", action)
+	}
+
+	// The opts argument tells us if the real rootfs is mounted and
+	// systemd is running normally.
+	// opts != nil: The direct rootfs path is supplied because the system is not
+	//              ready so we cannot use systemctl to modify unit files
+	//              without supplying the root path.
+	// opts == nil: No rootfs path is supplied because we are running with
+	//              rootfs mounted, so we can use systemctl normally. This
+	//              case is used for normal runtime changes.
+	var sysd systemd.Systemd
+	if opts != nil {
+		// Use systemctl for direct unit file manipulations (support a
+		// subset of unit operations such as enable/disable/mark/unmask)
+		// See: "systemctl --root=<dir>"
+		sysd = systemd.NewEmulationMode(opts.RootDir)
+	} else {
+		// Use systemctl with access to all the unit operations
+		sysd = systemd.New(systemd.SystemMode, &sysdCtrlAltDelLogger{})
+
+		// Make sure the ctrl-alt-del.target unit is in the expected state.
+		// (1) The required unit should be present (file exist under /lib/systemd/system).
```

Comment:
afaiu it's possible that the file can exists under /etc/systemd/system instead (thought that would be a rather odd setup) and our perception of systemd world will be unchanged
```suggestion
		// (1) The required unit should be present
```